themes/tdr/login/login.ftl

```diff
@@ -4,29 +4,42 @@
         ${msg("doLogIn")}
     <#elseif section = "form">
         <#if realm.password>
-            <form id="kc-form-login" onsubmit="login.disabled = true; return true;" action="${url.loginAction}" method="post">
-                <div class="govuk-form-group">
-                    <h1 class="govuk-label-wrapper">
-                        <label for="username" class="govuk-label govuk-label--l">
-                            ${msg("email")}
-                        </label>
-                    </h1>
-                    <input tabindex="1" id="username" class="govuk-input govuk-!-width-one-half" name="username" value="${(login.username!'')}"  type="text" autofocus autocomplete="off" />
-                </div>
+          <form id="kc-form-login" onsubmit="login.disabled = true; return true;" action="${url.loginAction}"
+                method="post">
+            <div class="govuk-form-group<#if message?has_content>--error</#if>">
+              <div class="govuk-form-group">
```

Comment:
Is the double nested `<div/> `necessary as they both have the same class `govuk-form-group`?

Reply:
The extra div is there so that vertical red error bar which shows on the left hand side covers both the username and password fields which is right. 
I did try with one div with the username and password fields inside it but it looks weird.

I also forgot to do the same thing with the form error for the reset password page which I've added in since you first looked at this.

Reply:
Fair enough.